tools/block_cache_trace_analyzer.cc

```diff
@@ -0,0 +1,325 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#include "tools/block_cache_trace_analyzer.h"
+
+#include <cinttypes>
+#include <set>
+#include "monitoring/histogram.h"
+
+namespace rocksdb {
+namespace {
+std::string block_type_to_string(TraceType type) {
+  switch (type) {
+    case kBlockTraceFilterBlock:
+      return "Filter";
+    case kBlockTraceDataBlock:
+      return "Data";
+    case kBlockTraceIndexBlock:
+      return "Index";
+    case kBlockTraceRangeDeletionBlock:
+      return "RangeDeletion";
+    case kBlockTraceUncompressionDictBlock:
+      return "UncompressionDict";
+    default:
+      break;
+  }
+  // This cannot happen.
+  assert(false);
+  return "";
```

Comment:
The assertion works only if compiled with debug level greater than 0. When debug level is 0, this function can return an empty string. Therefore, I am thinking maybe we can make this function return "InvalidType" or something like that instead of "".

Reply:
Thanks. I updated to return "InvalidType".